src/CountUp.tsx

```diff
@@ -172,30 +188,37 @@ class CountUp extends Component {
   start = () => {
     const { pauseResume, reset, restart: start, update } = this;
     const { delay, onEnd, onStart } = this.props;
-    const run = () =>
-      this.instance.start(() => onEnd({ pauseResume, reset, start, update }));
+    const run = () => {
+      this.instance?.start(() => {
+        if (onEnd !== undefined) {
+          onEnd({ pauseResume, reset, start, update });
+        }
+      });
+    };
 
     // Delay start if delay prop is properly set
-    if (delay > 0) {
-      this.timeoutId = setTimeout(run, delay * 1000);
+    if (delay && delay > 0) {
+      this.timeoutId = window.setTimeout(run, delay * 1000);
     } else {
       run();
     }
 
-    onStart({ pauseResume, reset, update });
+    if (onStart !== undefined) {
+      onStart({ pauseResume, reset, update });
+    }
```

Comment:
```suggestion
    onStart?({ pauseResume, reset, update });
```

Reply:
@glennreyes I'm not sure that it is valid TS code

Reply:
Ah, it should be `?.`. Feel free to change, thanks!